ggml/src/ggml-metal/ggml-metal.m

```diff
@@ -1126,6 +1126,13 @@ static bool ggml_metal_supports_op(const struct ggml_backend_metal_device_contex
         case GGML_OP_ARGMAX:
         case GGML_OP_NORM:
         case GGML_OP_ROPE:
+            const int mode = ((const int32_t *) op->op_params)[2];
+            if (mode && GGML_ROPE_TYPE_MROPE) {
+                return false;
+            }
+            if (mode && GGML_ROPE_TYPE_VISION) {
+                return false;
+            }
```

Comment:
Should add local scope here, otherwise the `mode` variable leaks to all other `switch` cases:

```suggestion
            {
	            const int mode = ((const int32_t *) op->op_params)[2];
	            if (mode && GGML_ROPE_TYPE_MROPE) {
	                return false;
	            }
	            if (mode && GGML_ROPE_TYPE_VISION) {
	                return false;
	            }
	        }
```

Same for the other backends.